connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -207,7 +208,7 @@ public ConnectorStateInfo connectorStatus(String connName) {
 
         Collections.sort(taskStates);
 
-        return new ConnectorStateInfo(connName, connectorState, taskStates);
+        return new ConnectorStateInfo(connName, connectorState, taskStates, null);
```

Comment:
If we pass `null` here, won't we never see the connector type in the output for the endpoint `/{connector}/status` since this is the method that gets called for that info and the result is just passed directly through `ConnectorsResource`?

Testing with this version of the PR, this seems to be the case:

```
$ curl http://localhost:8083/connectors/local-file-sink/status | jq
{
  "name": "local-file-sink",
  "connector": {
    "state": "RUNNING",
    "worker_id": "10.200.7.129:8083"
  },
  "tasks": [
    {
      "state": "RUNNING",
      "id": 0,
      "worker_id": "10.200.7.129:8083"
    }
  ],
  "type": null
}
```